daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/TypeCalculator.scala

```diff
@@ -429,10 +454,14 @@ object TypeCalculatorCompiler {
     }
   }
 
-  def compileExpression[A <: AnyRef, B <: AnyRef](optInputTypeCalc: Option[CompiledExpression[B]], optOutputTypeCalc: Option[CompiledExpression[A]], srcType: NodeInfo.Kind, dstType: NodeInfo.Kind): ExpressionTypeCalculator[A, B] = {
-    val maybeInputType: Maybe[CompiledExpression[B]] = optInputTypeCalc.map(Maybe(_)).getOrElse(Maybe.Nope)
-    val maybeOutputType: Maybe[CompiledExpression[A]] = optOutputTypeCalc.map(Maybe(_)).getOrElse(Maybe.Nope)
-    new ExpressionTypeCalculator(maybeInputType, maybeOutputType, srcType, dstType)
+  def compileExpression[A <: AnyRef, B <: AnyRef](
```

Comment:
I think we have too many overloads of compileExpression at this point. No reason to further proliferate that. Can this be called compileTypeCalcExpression ?